internal/webhooks/cluster.go

```diff
@@ -343,34 +353,65 @@ func (webhook *Cluster) validateTopology(ctx context.Context, oldCluster, newClu
 					fldPath.Child("version"),
 					oldCluster.Spec.Topology.Version,
-					fmt.Sprintf("old version %q cannot be compared with %q", oldVersion, inVersion),
+					"unable to parse old version",
 				),
 			)
 		}
-		if inVersion.NE(semver.Version{}) && oldVersion.NE(semver.Version{}) && version.Compare(inVersion, oldVersion, version.WithBuildTags()) == -1 {
-			allErrs = append(
-				allErrs,
-				field.Invalid(
-					fldPath.Child("version"),
-					newCluster.Spec.Topology.Version,
-					fmt.Sprintf("version cannot be decreased from %q to %q", oldVersion, inVersion),
-				),
-			)
-		}
-		// A +2 minor version upgrade is not allowed.
-		ceilVersion := semver.Version{
-			Major: oldVersion.Major,
-			Minor: oldVersion.Minor + 2,
-			Patch: 0,
-		}
-		if inVersion.GTE(ceilVersion) {
-			allErrs = append(
-				allErrs,
-				field.Forbidden(
-					fldPath.Child("version"),
-					fmt.Sprintf("version cannot be increased from %q to %q", oldVersion, inVersion),
-				),
-			)
+
+		if _, ok := newCluster.GetAnnotations()[clusterv1.ClusterTopologyUnsafeUpdateVersionValidationAnnotation]; ok {
+			log := ctrl.LoggerFrom(ctx)
+			log.Info("Skipping version validation for object because annotation is set", "annotation", clusterv1.ClusterTopologyUnsafeUpdateVersionValidationAnnotation)
+		} else {
+			// Version could only be increased.
+			if inVersion.NE(semver.Version{}) && oldVersion.NE(semver.Version{}) && version.Compare(inVersion, oldVersion, version.WithBuildTags()) == -1 {
+				allErrs = append(
+					allErrs,
+					field.Invalid(
+						fldPath.Child("version"),
+						newCluster.Spec.Topology.Version,
+						fmt.Sprintf("version cannot be decreased from %q to %q", oldVersion, inVersion),
+					),
+				)
+			}
+
+			// A +2 minor version upgrade is not allowed.
+			ceilVersion := semver.Version{
+				Major: oldVersion.Major,
+				Minor: oldVersion.Minor + 2,
+				Patch: 0,
+			}
+			if inVersion.GTE(ceilVersion) {
+				allErrs = append(
+					allErrs,
+					field.Forbidden(
+						fldPath.Child("version"),
+						fmt.Sprintf("version cannot be increased from %q to %q", oldVersion, inVersion),
+					),
+				)
+			}
+
+			// only check the following cases if the minor version increases by at least 1:
+			ceilVersion.Minor = oldVersion.Minor + 1
+			if inVersion.GTE(ceilVersion) {
+				// minor version cannot be increased if KCP is upgrading or not yet on the current version
+				if err := validateControlPlaneVersion(ctx, fldPath.Child("version"), webhook.Client, oldCluster, oldVersion); err != nil {
+					allErrs = append(allErrs, err)
+				}
+
+				// minor version cannot be increased if MachineDeployments are upgrading or not yet on the current version
+				if err := validateTopologyMachineDeploymentVersions(ctx, fldPath.Child("version"), webhook.Client, oldCluster, oldVersion); err != nil {
+					allErrs = append(allErrs, field.Invalid(fldPath.Child("version"), err, "blocking version update due to machine deployment version check"))
```

Comment:
```suggestion
					allErrs = append(allErrs, field.Invalid(fldPath.Child("version"), err, "blocking version update due to MachineDeployment version check"))
```
let's use the actual rersource names here (same below for MP)